src/components/navs/AppNav/AppNavClaimBtn.vue

```diff
@@ -0,0 +1,203 @@
+
+import { TOKENS } from '@/constants/tokens';
+import { bnum } from '@/lib/utils';
+import { claimRewards } from '@/services/claim';
+
+export default defineComponent({
+  name: 'AppNavClaimBtn',
+
+  setup() {
+    // DATA
+    const isClaiming = ref(false);
+    const rewardsEstimateSinceTimestamp = ref('0');
+
+    // COMPOSABLES
+    const { upToLargeBreakpoint } = useBreakpoints();
+    const store = useStore();
+    const userClaimsQuery = useUserClaimsQuery();
+    const { fNum } = useNumbers();
+    const { appNetwork, account } = useWeb3();
+    const { txListener } = useNotify();
+    const auth = useAuth();
+
+    const balPrice = computed(
+      () =>
+        store.state.market.prices[
+          getOriginalAddress(appNetwork.id, TOKENS.AddressMap.BAL)
+        ]?.price
+    );
+
+    // COMPUTED
+    const userClaims = computed(() =>
+      userClaimsQuery.isSuccess.value ? userClaimsQuery.data?.value : null
+    );
+
+    const availableToClaimInUSD = computed(() =>
+      balPrice.value != null && userClaims.value != null
+        ? bnum(userClaims.value?.availableToClaim)
+            .times(balPrice.value)
+            .toString()
+        : null
+    );
+
+    const currentRewards = computed(() =>
+      userClaims.value != null &&
+      userClaims.value.currentRewardsEstimate != null
+        ? bnum(userClaims.value.currentRewardsEstimate.rewards)
+            .plus(rewardsEstimateSinceTimestamp.value)
+            .toString()
+        : null
+    );
+
+    const currentRewardsInUSD = computed(() =>
+      balPrice.value != null && currentRewards.value != null
+        ? bnum(currentRewards.value)
+            .times(balPrice.value)
+            .toString()
+        : null
+    );
+
+    const totalRewards = computed(() =>
+      userClaims.value != null
+        ? bnum(userClaims.value.totalRewards)
+            .plus(rewardsEstimateSinceTimestamp.value)
+            .toString()
+        : null
+    );
+
+    useIntervalFn(async () => {
+      if (userClaims.value != null && userClaims.value.currentRewardsEstimate) {
+        const diffInSeconds = differenceInSeconds(
+          new Date(),
+          new Date(userClaims.value.currentRewardsEstimate.timestamp)
+        );
+        rewardsEstimateSinceTimestamp.value = bnum(diffInSeconds)
+          .times(userClaims.value.currentRewardsEstimate.velocity)
+          .toString();
+      }
+    }, 1000);
+
+    watch(account, () => {
+      rewardsEstimateSinceTimestamp.value = '0';
+    });
+
+    // METHODS
+    async function claimAvailableRewards() {
+      if (userClaims.value != null) {
+        isClaiming.value = true;
+        try {
+          const tx = await claimRewards(
+            appNetwork.id,
+            auth.web3,
+            account.value,
+            userClaims.value.pendingClaims,
+            userClaims.value.pendingClaimsReports
+          );
+          txListener(tx.hash);
+
+          await tx.wait();
```

Comment:
Blocknative updates too fast... I need to actually verify with Infura that the transaction was mined so that `userClaimsQuery` would actually return `0` when claimed.